[PWA-623]: Update Venia's inputs

anthoula opened the pull request:
## Description

Adjust input styles to match the new design

## Related Issue
https://jira.corp.magento.com/browse/PWA-623

## Acceptance 
* Verify inputs are updated throughout the app
* Ensure there is no broken functionality with the styling change

### Verification Stakeholders
@soumya-ashok 
@jimbo 
@dpatil-magento 

### Specification
<!-- Changes to `upward-spec` and/or `upward-js` packages must be reviewed -->
<!-- by `UPWARD-PHP` maintainers to ensure continued compatibility -->

### Verification Steps
<!-- Please describe in detail how a reviewer can verify your changes, -->
<!-- OR how you will demonstrate the changes to the stakeholder(s). -->
1. Navigate around the app following all known user flows
2. Verify all inputs previously present are present and work
3. Verify all inputs are positioned and relevant labels are readable

## Screenshots / Screen Captures (if appropriate)

## Checklist
<!--- Go over all the following points, and make sure you've done anything necessary -->
- [x] I have added tests to cover my changes, if necessary.
* I have updated the documentation accordingly, if necessary.

devops-pwa-codebuild commented:
**Performance Test Results**

The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.

https://pr-2510.pwa-venia.com : LH Performance Expected 0.85 Actual 0.5, LH Best Practices Expected 1 Actual 0.92, WPT Cache Expected 90 Actual 83.333333333333
https://pr-2510.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.35, LH Best Practices Expected 1 Actual 0.92
https://pr-2510.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.4, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92, WPT Cache Expected 65 Actual 50

jimbo commented on packages/venia-ui/lib/components/CartPage/ProductListing/quantity.css:
```diff
@@ -66,6 +61,15 @@
+}
+
+.button:disabled .icon svg,
+.button:disabled:hover .icon svg {
```
Ah, were we not able to pass a `--stroke` here for some reason?

anthoula replied:
Yeah, it's hard to pass in the custom variable for `--stroke` here, possibly due to the `.icon` override from above, which also declares a different value for `--stroke`

tjwiebell replied:
That icon override was overriding the `root` class with the `root` class from icon 🤯   Seemed redundant, so I deleted the override and was able to get variable cascade to work as expected.

```css
.button:hover {
    --stroke: rgb(var(--venia-global-color-gray-900));
}

.button:disabled,
.button:disabled:hover {
    --stroke: rgb(var(--venia-global-color-gray-400));
}
```

jimbo commented on packages/venia-ui/lib/components/RadioGroup/radio.js:
```diff
@@ -13,24 +14,30 @@ export class RadioOption extends Component {
                     {...rest}
                     className={classes.input}
                     id={id}
+                    defaultChecked={initialChecked}
```
Good catch.

tjwiebell replied:
A radio should get its checked state from the parent RadioGroup value. Just curious why this was needed.

jimbo commented on packages/venia-ui/lib/components/SearchBar/autocomplete.css:
```diff
@@ -16,15 +16,18 @@
     padding: 1rem 1rem;
     position: absolute;
     right: 0;
+    top: 2.2rem;
```
Can this be rounded to an integer?

```suggestion
    top: 2.25rem;
```

jimbo commented on packages/venia-ui/lib/index.css:
```diff
@@ -2,6 +2,8 @@
 :global(*) {
     box-sizing: border-box;
+    margin: 0;
+    padding: 0;
```
I used this shortcut on the styleguide app, but this seems like something that could cause widespread or hard-to-find regressions. Does it seem otherwise to you?

anthoula replied:
This was to fix an issue with radio buttons positioning. This looked fine in multiple browsers. However, to be safe, I will move this reset into the RadioGroup component.

soumya-ashok commented:
@jimbo @anthoula 

On first glance, looking at the fields, particularly on pages like checkout where there are a number of them - the combination of the 2px outline and the darker gray color feels visually overwhelming to me.

We could keep the gray and bump the outline down to 1px or, keep the outline at 2px but change the color to Hex #B5B8BA.

<img width="679" alt="Screen Shot 2020-07-08 at 9 47 14 AM" src="https://user-images.githubusercontent.com/25960021/86933675-6791f180-c100-11ea-997c-06176d110f62.png">

I have the same comment about the search field (I know we addressed that as a unique case), but an additional thing there is that the text "search" in the header turns a blue that isn't in our palette when the search field is engaged.

<img width="983" alt="Screen Shot 2020-07-08 at 9 25 06 AM" src="https://user-images.githubusercontent.com/25960021/86933788-8c866480-c100-11ea-80d8-8509e78d205c.png">

anthoula commented:
> I have the same comment about the search field (I know we addressed that as a unique case), but an additional thing there is that the text "search" in the header turns a blue that isn't in our palette when the search field is engaged.

@soumya-ashok this was actually introduced in another change and is present on develop: https://develop.pwa-venia.com

anthoula commented:
> looks like the Account sign in and creation forms are still reflecting the old style of required fields.

> The field "street address 2" is an optional field, but is currently not reflecting the optional field style that is in the XD file.

@soumya-ashok Good catch. I've updated the areas to remove required symbols, and added optional symbols where applicable

soumya-ashok commented:
@anthoula Thanks for making the spacing adjustments, the field dense forms are looking better!

Per discussion, the field borders will be adjusted to be a lighter gray as part of an accessibility story

I found a few things I wanted to bring to your attention.

The header icons, which are supposed to be the darkest gray are now all a lighter gray - not sure why this happened but thought it may be a result of us trying to adjust the border color on fields.

The ZIP code field in shipping estimation on the cart works even when the field value is wrong - I tried a 4 digit zip and it still gave me the estimate

I'm seeing a white background on checkboxes that are particularly evident in the focus state

<img width="229" alt="Screen Shot 2020-07-09 at 1 47 46 PM" src="https://user-images.githubusercontent.com/25960021/87079519-cb451900-c1eb-11ea-9938-222ac3b34092.png">

anthoula commented:
> The ZIP code field in shipping estimation on the cart works even when the field value is wrong - I tried a 4 digit zip and it still gave me the estimate

@soumya-ashok this was actually introduced in another change and is present on develop: https://develop.pwa-venia.com

soumya-ashok commented:
@anthoula The changes to inputs are UX approved for this round - We'll come back and adjust the grays as a follow up. Thanks for the great work!

jimbo commented:
This looks good to me. I'm sure there'll be a few edges to find in QA; I'll do some QA myself in the meantime.

tjwiebell commented on packages/venia-ui/lib/components/ForgotPassword/ForgotPasswordForm/__tests__/__snapshots__/forgotPasswordForm.spec.js.snap:
```diff
@@ -22,6 +21,7 @@ exports[`renders correctly 1`] = `
       <span>
         <input
           autoComplete="email"
+          className="undefined "
```
Consider mocking classify here too.

tjwiebell commented on packages/venia-ui/lib/components/RadioGroup/radio.js:
```diff
@@ -13,24 +14,30 @@ export class RadioOption extends Component {
                     className={classes.input}
                     id={id}
+                    defaultChecked={initialChecked}
+                    type="radio"
```
Was this prop needed?

https://github.com/joepuzzo/informed/blob/master/src/components/form-fields/Radio.js#L49

anthoula replied:
It's redundant, good call

jimbo commented on packages/venia-ui/lib/components/CartPage/GiftCards/giftCards.js:
```diff
@@ -122,7 +122,11 @@ const GiftCards = props => {
-                <div className={classes.card_input_container}>
+                <div
+                    className={`${classes.card_input_container} ${
+                        shouldDisplayCardError ? classes.error : ''
```
@tjwiebell No, I definitely overlooked it. You're right, we avoid classname concatenation in all cases.

As for your suggestion, I don't recommend using the `:not` pseudo selector.

anthoula replied:
All cases should now use modifiers, with the exception of `filterDefault.js`. Those checkboxes should really be `<input type="checkbox">`, instead of `<button>`. Semantics of filters will be addressed in upcoming story PWA-713.

tjwiebell commented on packages/venia-ui/lib/components/TextInput/textInput.js:
```diff
@@ -30,13 +30,17 @@ export class TextInput extends Component {
+        const inputClass = fieldState.error
+            ? classes.input_error
+            : classes.input;
+        console.log(inputClass);
```
```suggestion
```

dpatil-magento commented:
@jimbo   we see minor issue in Android device where Radios in Qty drop down, County/State in shipping form dropdown does not look same as rest of site.  But its very specific to how android handles it, iOS don't have issue as there is no Radios on these fields. Does this needs to be handled?

cc @anthoula 

![image](https://user-images.githubusercontent.com/40405790/87451091-0321d700-c5c5-11ea-93a5-6fcde356517e.png)

jimbo commented:
> @jimbo we see minor issue in Android device where Radios in Qty drop down, County/State in shipping form dropdown does not look same as rest of site. But its very specific to how android handles it, iOS don't have issue as there is no Radios on these fields. Does this needs to be handled?
> 
> cc @anthoula
> 
> ![image](https://user-images.githubusercontent.com/40405790/87451091-0321d700-c5c5-11ea-93a5-6fcde356517e.png)

That's working as intended. No fix necessary. 👍

dpatil-magento commented:
QA Approved.
